keygen.go

```diff
@@ -180,12 +184,36 @@ func (keygen *KeyGen) generateEc(kt KeyType) (crypto.PrivateKey, error) {
 	return ecdsa.GenerateKey(curve, keygen.rng)
 }
 
+func (keygen *KeyGen) generate25519(kt KeyType) (crypto.PrivateKey, error) {
+	switch kt {
+	case X25519:
+		var privKey [32]byte
+		_, err := io.ReadFull(keygen.rng, privKey[:])
+
+		// from https://cr.yp.to/ecdh.html
+		privKey[0] &= 248
+		privKey[31] &= 127
+		privKey[31] |= 64
+
+		return x25519PrivateKey(privKey[:]), err
+	case ED25519:
+		_, privKey, err := ed25519.GenerateKey(keygen.rng)
+		return &privKey, err
+	}
+
+	return nil, errors.New("invalid key type requested")
+}
+
 func (keygen *KeyGen) GenerateKey(kt KeyType) (crypto.PrivateKey, error) {
 	switch kt {
 	case EC256, EC521:
 		return keygen.generateEc(kt)
 	case RSA2048, RSA4096:
 		return keygen.generateRsa(kt)
+	case X25519:
+		fallthrough
```

Comment:
why fallthrough here and not just do `case X25519, ED25519`

Reply:
dunno - maybe aliens have stolen my brain when I was doing this

Reply:
fixed